lib/proxy_api/proxy_exception.rb

```diff
@@ -3,12 +3,20 @@ class ProxyException < ::Foreman::WrappedException
     attr_reader :url
 
     def initialize(url, exception, message, *params)
+      message += append_exception_message(exception)
```

Comment:
Each proxy feature should already have a unique set of error codes (e.g. ProxyExceptions are unique within each ProxyAPI class), but I agree there are different errors that the error code wouldn't reflect.

I'm concerned though that the text from the proxy may not be standardised - if it's used to generate the error code then the codes may not be reproducible at all, which is probably worse than them being too broad.

e.g. if the proxy returns an error stating it can't contact 192.168.0.100, then an error code generated based on that IP isn't useful to anybody except that one user, or ditto with file paths (cannot open file /srv/tftpboot/ or /var/lib/tftpboot).